drivers/ad5683/ad5683.c

```diff
@@ -0,0 +1,334 @@
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "ad5683.h"
+#include "platform_drivers.h"
+#include <stdint.h>
+#include <stdlib.h>
+
+/*****************************************************************************/
+/***************************** Constant definition ***************************/
+/*****************************************************************************/
+
+static const struct ad5683_chip_info chip_info[] = {
+	[ID_AD5681R] = {
+		.resolution = 12,
+		.communication = SPI,
+	},
+	[ID_AD5682R] = {
+		.resolution = 14,
+		.communication = SPI,
+	},
+	[ID_AD5683R] = {
+		.resolution = 16,
+		.communication = SPI,
+	},
+	[ID_AD5683] = {
+		.resolution = 16,
+		.communication = SPI,
+	},
+	[ID_AD5691R] = {
+		.resolution = 12,
+		.communication = I2C,
+	},
+	[ID_AD5692R] = {
+		.resolution = 14,
+		.communication = I2C,
+	},
+	[ID_AD5693R] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_AD5693] = {
+		.resolution = 16,
+		.communication = I2C,
+	}
+};
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/**
+ * @brief Initializes the communication peripheral and the initial Values for
+ *        AD5683 Board.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return ret - The result of the initialization procedure.
+ *               Example: -1 - SPI peripheral was not initialized or the
+ *                             device is not present.
+ *                         0 - SPI peripheral was initialized and the
+ *                             device is present.
+*/
+int32_t ad5683_init(struct ad5683_dev **device,
+		    struct ad5683_init_param init_param)
+{
+	struct ad5683_dev *dev;
+	int32_t ret;
+
+	dev = (struct ad5683_dev *)malloc(sizeof(*dev));
```

Comment:
would `calloc()`  be a better idea here ?
calloc() guarantees that the memory s initialized to 0; malloc() doesn't
i agree that there are many places where malloc() is used (though calloc()) would be better suited ;

Reply:
The structure is initialized afterwards in the same function with init_params. Does it make sense to initialize it twice? Moreover, I think `malloc` is a bit faster that `calloc`.

Reply:
agreed; malloc() is faster ; but calloc() is less error prone ;
i admit i haven't looked at the init of the device, but, if it's fully initialized, and no field member is omitted, then malloc() is fine